src/RelayClient.ts

```diff
@@ -48,16 +80,188 @@ class RelayClient extends EnvelopingEventEmitter {
     this._envelopingConfig = getEnvelopingConfig();
   }
 
-  public _getRelayRequestDetails = async (
-    // FIXME: make private
-    _request: UserDefinedRelayRequest
-  ): Promise<RelayRequest> => {
-    return await Promise.resolve({} as EnvelopingTypes.RelayRequestStruct);
+  public estimateTokenTransferGas = async (
+    _props: TokenGasEstimationParams
+  ): Promise<BigNumber> => {
+    return Promise.resolve(constants.Zero);
   };
 
-  // TODO: lift the ts exception once method is used
   // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-  // @ts-ignore
+  // @ts-ignore FIXME: remove the ts ignore once the method is consumed
+  private _getEnvelopingRequestDetails = async (
+    envelopingRequest: UserDefinedEnvelopingRequest,
+    {
+      isSmartWalletDeploy,
+      preDeploySWAddress,
+      forceGasLimit,
+      forceGasPrice,
+      internalEstimationCorrection,
+      addExternalCorrection: addCorrection,
+    }: RequestConfig
+  ): Promise<EnvelopingRequest> => {
+    const isDeployment: boolean = isDeployRequest(
+      envelopingRequest as EnvelopingRequest
+    );
+    const {
+      relayData: { callForwarder },
+      request: { data, from, to, tokenContract },
+    } = envelopingRequest;
+
+    if (!callForwarder) {
+      throw new Error(MISSING_CALL_FORWARDER);
+    }
+
+    const callVerifier: PromiseOrValue<string> =
+      envelopingRequest.relayData.callVerifier ||
+      this._envelopingConfig[
+        isDeployment ? 'deployVerifierAddress' : 'relayVerifierAddress'
+      ];
+
+    if (!callVerifier) {
+      throw new Error('No call verifier present. Check your configuration.');
+    }
+
+    const relayServerUrl = this._envelopingConfig.preferredRelays.at(0)?.url;
+
+    if (!relayServerUrl) {
+      throw new Error(
+        'Check that your configuration contains at least one preferred relay with url.'
+      );
+    }
+    const feesReceiver: PromiseOrValue<string> = (
+      await new HttpClient().getChainInfo(relayServerUrl)
+    ).feesReceiver;
+
+    if (!feesReceiver) {
```

Comment:
Are we going to keep like this or are you going to change it or should I change it once I need it?

Reply:
changed now